cli/src/commands/validate/validate.ts

```diff
@@ -3,39 +3,56 @@ import { existsSync, promises as fs, readFileSync, readdirSync, statSync } from
 import pkg from '@stoplight/spectral-core';
 const { Spectral } = pkg;
 import { getRuleset } from '@stoplight/spectral-cli/dist/services/linter/utils/getRuleset.js';
+import * as winston from 'winston';
 
-export default async function validate(jsonSchemaInstantiationLocation: string, jsonSchemaLocation: string, metaSchemaPath: string) {
+let logger: winston.Logger; // defined later at startup
+
+export default async function validate(jsonSchemaInstantiationLocation: string, jsonSchemaLocation: string, metaSchemaPath: string, debug: boolean) {
+    initLogger(debug);
     let exitCode = 0;
     try {
         const ajv = new Ajv2020({ strict: false });
 
         loadMetaSchemas(ajv, metaSchemaPath);
 
-        console.info(`Loading pattern from : ${jsonSchemaLocation}`);
+        logger.info(`Loading pattern from : ${jsonSchemaLocation}`);
         const jsonSchema = await getFileFromUrlOrPath(jsonSchemaLocation);
 
-        console.info(`Loading pattern instantiation from : ${jsonSchemaInstantiationLocation}`);
+        logger.info(`Loading pattern instantiation from : ${jsonSchemaInstantiationLocation}`);
         const jsonSchemaInstantiation = await getFileFromUrlOrPath(jsonSchemaInstantiationLocation);
 
         const validateSchema = ajv.compile(jsonSchema);
 
         if (!validateSchema(jsonSchemaInstantiation)) {
-            console.error('The instantiation does not match the JSON schema pattern. Errors: ', validateSchema.errors);
+            logger.error(`The instantiation does not match the JSON schema pattern. Errors: ${prettifyJson(validateSchema.errors)}'`);
             exitCode = 1;
         } else {
-            console.info('The schema instantiation matches the json schema');
+            logger.info('The schema instantiation matches the json schema');
         }
 
         await runSpectralValidations(jsonSchemaInstantiation);
     } catch (error) {
-        console.error(`An error occured: ${error}`);
+        logger.error(`An error occured: ${error}`);
         process.exit(1);
     }
     process.exit(exitCode);
 }
 
+function initLogger(debug: boolean): void {
+    const level = debug ? 'debug' : 'info';
+    logger = winston.createLogger({
+        transports: [
+            new winston.transports.Console()
+        ],
+        level: level,
+        format: winston.format.combine(
+            winston.format.cli(),
```

Comment:
Why the combine()? Are you not just using one format?

Also - is there any chance that we could re-use this code across the three modules?

Reply:
Good point, I was trying to see if I could print JSON using a different format I forgot to remove the combine. Should I create and helper.ts file where I can add this logic so that it can be reused across the three commands? @willosborne

Reply:
I think this is a good approach yes